src/tools/tracejob.c

```diff
@@ -382,21 +382,39 @@ void
 parse_log(FILE *fp, char *job, int ind)
 {
 	struct log_entry tmp;	/* temporary log entry */
-	char buf[16384];	/* buffer to read in from file */
+	char *buf;		/* buffer to read in from file */
+	char *tbuf;		/* help wth larger entries in log */
 	char job_buf[128];	/* hold the jobid and the . */
 	char *p;		/* pointer to use for strtok */
 	int field_count;	/* which field in log entry */
 	int j = 0;
-	struct tm tms;	/* used to convert date to unix date */
+	struct tm tms;		/* used to convert date to unix date */
 	int lineno		= 0;
 	int slen;
 	char *pdot;
+	int buf_size = 16384;	/* initial buffer size */
+	int break_fl = 0;
+
+	buf = (char*)malloc(buf_size * sizeof(char));
+	tbuf = (char*)malloc(buf_size * sizeof(char));
 
 	tms.tm_isdst = -1;	/* mktime() will attempt to figure it out */
 
 	strcpy(job_buf, job);
 
-	while (fgets(buf, 16384, fp) != NULL) {
+	while (fgets(buf, buf_size, fp) != NULL) {
+		while(buf_size == strlen(buf) + 1){
+			buf = (char*)realloc(buf, buf_size * sizeof(char*));
+			tbuf = (char*)realloc(tbuf, buf_size * sizeof(char*));
+			if(buf == NULL || tbuf == NULL || fgets(tbuf, buf_size+1, fp) == NULL){
```

Comment:
I think your +1 will run you off the end of your array.  Since tbuf is buf_size in length, fgets() will read len-1 bytes and put a '\0' at the end.  Since you are reading buf_size+1, it'll read buf_size bytes and put a '\0' in non-allocated space.